File: backend/src/main/java/com/suitcustom/entity/Customization.java
package com.suitcustom.entity;

import lombok.Data;
import org.apache.ibatis.type.JdbcType;
import com.suitcustom.handler.JsonTypeHandler;
import java.io.Serializable;
import java.util.Date;
import java.util.Map;
import java.math.BigDecimal;

/**
 * 定制实体类
 * 
 * @author suitcustom
 */
@Data
public class Customization implements Serializable {

  private static final long serialVersionUID = 1L;

  /**
   * ID
   */
  private Long id;

  /**
   * 定制编号(格式：CUST202503007)
   */
  private String code;

  /**
   * 用户ID
   */
  private Long userId;

  /**
   * 款式ID
   */
  private Long styleId;

  /**
   * 面料ID
   */
  private Long fabricId;

  /**
   * 定制名称
   */
  private String name;

  /**
   * 定制价格
   */
  private BigDecimal price;

  /**
   * 尺寸信息(JSON)
   */
  private Map<String, Object> sizeInfo;

  /**
   * 定制选项(JSON)
   */
  private Map<String, Object> options;

  /**
   * 收货人信息
   */
  private Map<String, String> receiverInfo;

  /**
   * 备注
   */
  private String remark;

  /**
   * 状态(0:草稿,1:待确认,2:生产中,3:已完成)
   */
  private Integer status;

  /**
   * 进度(0-100)
   */
  private Integer progress;

  /**
   * 创建时间
   */
  private Date createTime;

  /**
   * 更新时间
   */
  private Date updateTime;
}

File: frontend/customer-web/src/types/customization.ts
import type { StyleInfo } from './style'
import type { FabricInfo } from './fabric'

export interface CustomizationInfo {
  id: number
  code: string
  userId: number
  styleId: number
  fabricId: number
  style?: StyleInfo
  fabric?: FabricInfo
  sizeInfo: Record<string, number>
  receiverInfo?: Record<string, string>
  price: number
  status: number
  progress: number
  remark?: string
  createTime: string
  updateTime: string
}

export interface SizeInfo {
  height: number // 身高
  weight: number // 体重
  shoulder: number // 肩宽
  bust: number // 胸围
  waist: number // 腰围
  hip: number // 臀围
  sleeve: number // 袖长
  thigh: number // 大腿围
  length: number // 衣长
}

export interface ReceiverInfo {
  name: string
  phone: string
  address: string
} 